src/libs/tmdk/tests/deploy.test.ts

```diff
@@ -1,26 +1,265 @@
+/* eslint @typescript-eslint/no-var-requires: "off" */
 // Copyright 2022 Amazon.com, Inc. or its affiliates. All Rights Reserved.
 // SPDX-License-Identifier: Apache-2.0
 
 import { handler, Options } from "../src/commands/deploy";
 import { Arguments } from "yargs";
-import { mockExit } from "./test-utils";
+import {
+  workspaceId,
+  mockExit,
+  s3BucketArn,
+  s3ContentLocationBase,
+  workspaceBucket,
+} from "./test-utils";
+import { mockClient } from "aws-sdk-client-mock";
+import {
+  CreateComponentTypeCommand,
+  CreateEntityCommand,
+  CreateSceneCommand,
+  GetComponentTypeCommand,
+  GetEntityCommand,
+  GetWorkspaceCommand,
+  IoTTwinMakerClient,
+  ResourceNotFoundException,
+} from "@aws-sdk/client-iottwinmaker";
+import {
+  emptyTmdk,
+  componentType1Definition,
+  scene1,
+  entity1Definition,
+} from "./test-constants";
+import { PutObjectCommand, S3Client } from "@aws-sdk/client-s3";
+import * as fsPromises from "fs/promises";
+const fs = require("fs");
 
-jest.mock("../src/lib/aws-clients");
+jest.mock("fs/promises", () => {
+  return { readFile: jest.fn() };
+});
+const twinmakerMock = mockClient(IoTTwinMakerClient);
+const s3Mock = mockClient(S3Client);
+const fakeTmdkDir = "/tmp/deploy-unit-tests";
 
 describe("testing deploy", () => {
-  beforeEach(mockExit.mockClear);
+  beforeEach(() => {
+    twinmakerMock.reset();
+    s3Mock.reset();
+    jest.resetAllMocks();
+    mockExit.mockClear();
+  });
 
   test("deploy_givenNoTMDKProject_expectError", async () => {
```

Comment:
prefer test names such as 'test('throws error when given directory that does not exist')' i.e. human readable test descriptions in the for mat of "does X when Y"

Reply:
Got it, will update